### PR TITLE
Add relatedChart field in chart config to indicate whether related chart can be scaled

### PR DESCRIPTION
**Pull request by shifucun**
relatedChart field contains the information for derived chart. It contains "scale" and "denominator" fields to indicate how scaling should be done

### CODE REVIEW

**Comment by beets** [comment]
i wonder if we should instead explicitly define a denominator (and if not defined, do not scale the chart). we often scale by population, when we should scale by a subpopulation.

**Comment by shifucun** [comment]
Added a few denominator cases, otherwise default to total population

File: static/js/place/place_template.tsx [hunk 1]
```diff
@@ -556,8 +557,18 @@ class ChartBlock extends Component<ChartBlockPropType, unknown> {
     );
   }
 
-  // TODO(shifucun): Add more config to indicate whether to use perCapita for
-  // place comparison.
+  private copyAndUpdateConfig(config: ConfigType) {
+    const conf = { ...config };
+    conf.chartType = chartTypeEnum.GROUP_BAR;
+    conf.axis = "PLACE";
+    if (conf.canScale == null || conf.canScale) {
```
Comment (beets) [inline static/js/place/place_template.tsx:564]: this should update if there conf.denominator is defined (or we can revert the extra canScale from the config - i prefer this option)

File: static/js/place/place_template.tsx [hunk 2]
```diff
@@ -568,9 +579,7 @@ class ChartBlock extends Component<ChartBlockPropType, unknown> {
     conf.title = conf.title + " in " + this.props.placeName;
     result.push(conf);
 
-    conf = { ...config };
-    conf.chartType = chartTypeEnum.GROUP_BAR;
-    conf.axis = "PLACE";
+    conf = this.copyAndUpdateConfig(config);
```
Comment (beets) [inline static/js/place/place_template.tsx:582]: i missed this in my last PR - could you also use displayPlaceType for the else clause (line 588)

File: server/chart_config.json [hunk 3]
```diff
@@ -35,40 +33,51 @@
...
       "UnemploymentRate_Person_Female"
-    ]
+    ],
+    "canScale": false
   },
   {
     "category": ["Economics", "Economic Activity"],
     "title": "Gross Domestic Product",
     "statsVars": ["Amount_EconomicActivity_GrossDomesticProduction_Nominal"],
+    "unit": "$",
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/chart_config.json:56]: Use relatedChart field as discussed

File: static/js/place/place_template.tsx (hunk 2 above)
Comment (shifucun) [reply static/js/place/place_template.tsx:578]: it should be okay there now

**Comment by shifucun** [comment]
ptal with the new change

**Review by beets — COMMENTED** [review]
i really like this new format - very easy to follow along

the data_fetcher code expects denominator, if specified, to be a string[] - will that be updated in a followup? a shortcut would be to expand that from a string to an array (of the right length) when copying the config.

**Comment by shifucun** [comment]
> i really like this new format - very easy to follow along
> 
> 
> 
> the data_fetcher code expects denominator, if specified, to be a string[] - will that be updated in a followup? a shortcut would be to expand that from a string to an array (of the right length) when copying the config.

Expanding string to a list sounds good

**Review by beets — APPROVED** [review]

**Pull request merged**